src/libos/src/fs/file_table.rs

```diff
@@ -65,9 +65,20 @@ impl FileTable {
         Ok(min_free_fd)
     }
 
-    pub fn put(&mut self, file: FileRef, close_on_spawn: bool) -> FileDesc {
+    pub fn put(&mut self, file: FileRef, close_on_spawn: bool) -> Result<FileDesc> {
         let mut table = &mut self.table;
 
+        let soft_rlimit_nofile = current!()
```

Comment:
I find that the "fn dup" at line 28 and "fn put_at" at line 100 need a similar check.